worker/src/lib.rs

```diff
@@ -299,7 +300,73 @@ pub async fn witness_discord(
             vc.generate_proof(
                 &sk,
                 &LinkedDataProofOptions {
-                    verification_method: Some(format!("{}#controller", SPRUCE_DIDWEB)),
+                    verification_method: Some(URI::String(format!("{}#controller", SPRUCE_DIDWEB))),
+                    ..Default::default()
+                }
+            )
+            .await
+        );
+        vc.proof = Some(OneOrMany::One(proof));
+
+        Ok(jserr!(serde_json::to_string(&vc)).into())
+    })
+}
+
+#[wasm_bindgen]
+pub async fn dns_lookup(
+    secret_key_jwk: String,
+    public_key_tezos: String,
+    domain: String,
+) -> Promise {
+    initialize_logging();
+
+    future_to_promise(async move {
+        let pk: JWK = jserr!(jwk_from_tezos_key(&public_key_tezos));
+        let sk: JWK = jserr!(serde_json::from_str(&secret_key_jwk));
+
+        let dns_result = jserr!(dns::retrieve_dns_response(domain.clone()).await);
+
+        let mut vc = jserr!(dns::build_dns_vc(&pk));
+
+        let mut signature_to_resolve = "".to_string();
+        for answer in dns_result.Answer {
+            let trimmed_signature: &str = &answer.data[1..answer.data.len() - 1];
+            if trimmed_signature.starts_with("TZP") {
+                signature_to_resolve = trimmed_signature.to_string();
+            }
+        }
+
+        let (sig_target, sig) = jserr!(dns::extract_dns_signature(signature_to_resolve));
+
+        jserr!(verify_signature(&sig_target, &pk, &sig));
+
+        let mut props = HashMap::new();
+        props.insert(
+            "publicKeyJwk".to_string(),
+            jserr!(serde_json::to_value(pk.clone())),
+        );
+
+        let mut evidence_map = HashMap::new();
+
+        evidence_map.insert("domain".to_string(), serde_json::Value::String(domain));
```

Comment:
I'm really not sure what to put as evidence, maybe we don't put anything (apart from the timestamp) because there's nothing that would help finding the record again.

Reply:
Removed

Reply:
I would still recommend raising it with the others